contrib/google.golang.org/grpc/option.go

```diff
@@ -68,6 +78,21 @@ func NoDebugStack() Option {
 	}
 }
 
+// NonErrorCodes marks the given codes to be treated as non-errors.
+// codes.OK will always be treated as a success.
+func NonErrorCodes(cs ...codes.Code) InterceptorOption {
+	return func(cfg *config) {
+		nonErrorCodes := make(map[codes.Code]bool, len(cs)+1)
```

Comment:
I think all that we need here is:
```go
if cfg.nonErrorCodes == nil {
    cfg.nonErrorCodes = make(map[codes.Code]bool, len(cs))
}
for _, c := range cs {
    cfg.nonErrorCodes[c] = true
}
```
The rest are already set in `defaults`. Even the `nil` check would not really be necessary because it is assumed that config is initialized. But better safe than sorry. I'll let you decide if you want to keep it or not.

Reply:
Hmm... But what if someone wants to make `codes.Canceled` to be considered an error?
I for one have a few services where I'd like to log those as errors in APM.

Reply:
Alternatively we could also make it so that `default()` doesn't initialize the `nonErrorCodes` member. If `nonErrorCodes == nil` we could then use the old logic in `finishWithError()`.
This would allow us to use the simplified code you suggested above AND have proper backwards compatibility.

Reply:
Oh and again: If you really want to do it the way you suggested it (without being able to change the `codes.Canceled` bahavior), just say so and I'll change this.

Reply:
I guess we could do that, just use the `NonErrorCodes` option to override all the settings (except OK obviously). In that case we'd have something like this:

```go
cfg.nonErrorCodes = make(map[codes.Code]bool, len(cs))
for _, c := range cs {
    cfg.nonErrorCodes[c] = true
}
```

Then we should make it a bit more clear in the documentation, maybe have something like:

```go
// NonErrorCodes determines the list of codes which will not be considered errors in instrumentation.
// This call overrides the default handling of codes.Canceled as a non-error.
```